Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -666,66 +666,35 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
         )
     }
 
-    func schemeExecutionAction(action: ExecutionAction,
-                               graphTraverser: GraphTraversing,
-                               generatedProjects: [AbsolutePath: GeneratedProject],
-                               rootPath _: AbsolutePath) throws -> XCScheme.ExecutionAction
-    {
+    func schemeExecutionAction(
+        action: ExecutionAction,
+        graphTraverser: GraphTraversing,
+        generatedProjects: [AbsolutePath: GeneratedProject],
+        rootPath: AbsolutePath
+    ) throws -> XCScheme.ExecutionAction {
         guard
             let targetReference = action.target,
-            let graphTarget = graphTraverser.target(path: targetReference.projectPath, name: targetReference.name),
-            let generatedProject = generatedProjects[graphTarget.project.xcodeProjPath]
+            let graphTarget = graphTraverser.target(path: targetReference.projectPath, name: targetReference.name)
         else {
-            return schemeExecutionAction(action: action)
+            return XCScheme.ExecutionAction(
+                scriptText: action.scriptText,
+                title: action.title,
+                environmentBuildable: nil
+            )
         }
-        return schemeExecutionAction(
-            action: action,
-            target: graphTarget.target,
-            generatedProject: generatedProject
-        )
-    }
 
-    private func schemeExecutionAction(action: ExecutionAction) -> XCScheme.ExecutionAction {
-        XCScheme.ExecutionAction(
-            scriptText: action.scriptText,
-            title: action.title,
-            environmentBuildable: nil
+        let buildableReference = try createBuildableReference(
+            graphTarget: graphTarget,
+            graphTraverser: graphTraverser,
+            rootPath: rootPath,
+            generatedProjects: generatedProjects
```

Comment:
We already had a method to deal with resolving the correct buildable reference - I removed some of the redundant code around this area and re-used it resolve the issue 🙂